tensorboard/webapp/metrics/effects/index.ts

```diff
@@ -356,26 +357,29 @@ export class MetricsEffects implements OnInitEffects {
     })
   );
 
-  private readonly removeSavedPinsOnDisable$ = this.actions$.pipe(
+  private readonly addOrRemovePinsOnToggle$ = this.actions$.pipe(
     ofType(actions.metricsEnableSavingPinsToggled),
     withLatestFrom(
+      this.store.select(selectors.getPinnedCardsWithMetadata),
       this.store.select(selectors.getEnableGlobalPins),
       this.store.select(selectors.getShouldPersistSettings),
       this.store.select(selectors.getMetricsSavingPinsEnabled)
     ),
     filter(
-      ([
-        ,
-        enableGlobalPins,
-        getShouldPersistSettings,
-        getMetricsSavingPinsEnabled,
-      ]) =>
-        enableGlobalPins &&
-        getShouldPersistSettings &&
-        !getMetricsSavingPinsEnabled
+      ([, , enableGlobalPins, getShouldPersistSettings]) =>
+        enableGlobalPins && getShouldPersistSettings
     ),
-    tap(() => {
-      this.savedPinsDataSource.removeAllScalarPins();
+    tap(([, pinnedCards, , , getMetricsSavingPinsEnabled]) => {
+      if (getMetricsSavingPinsEnabled) {
+        const tags = pinnedCards
```

Comment:
nit: type assertions (`...as Tag[]`) are unsafe and (in principle) can cause runtime crashes.

Can you try: `const tags: Tag[] = ...`. You may also need to type the filter function callback: `filter((v): v is Tag => v !== null)`

Reply:
Applied! Thanks !